service/invoker/invoker.go

```diff
@@ -177,3 +187,14 @@ func (i *Invoker) Script(height uint64, script []byte, arguments []cadence.Value
 
 	return proc.Value, nil
 }
+
+func (i *Invoker) validateHeightIndexed(height uint64) error {
```

Comment:
given that we duplicate this code, should it be a part of the `index` access itself?  Otherwise index usage would be quite error-prone and involve not forgetting to call `validateHeightIndexed`.

Reply:
So I thought of this earlier and had two problems with it:
- `ReadRegister`, the function we want to guard for script exec is not in the `index` module: 
https://github.com/onflow/flow-archive/blob/2be1d7f579efc8d16dcd9ca2b8c8cec4611a596a/service/invoker/read.go
- for the index functions, I wasn't sure if we should implement this guard for every lookup by height, including `SealsByHeight`, `TransactionsByHeight`, `CollectionsByHeight` etc. 

So I went with this duplicating approach to only guard where relevant to avoid more surgery. 

That said, I've moved it to a util so we can use it in `index` if we decide to go that way later on